listenbrainz/webserver/views/api_tools.py

```diff
@@ -48,7 +51,7 @@ def insert_payload(payload, user, listen_type=LISTEN_TYPE_IMPORT):
         raise
     except Exception as e:
         print(e)
-    return augmented_listens
+    return RedisListenStore.store_latest_listens(augmented_listens)
```

Comment:
we should do something like this here.
```suggestion
    redis_connection._redis.store_latest_listens(payload)
    return augmented_listens
```